MIPs/mip-x-multichainapi.md

```diff
@@ -0,0 +1,236 @@
+Multichain API connections will be established and updated through [CAIP-25](https://chainagnostic.org/CAIPs/caip-25) `wallet_createSession` calls.
+
+> **Note:** A `sessionId` will not be returned in the initial response. Instead, the API will adopt session lifecycle management methods outlined in [CAIP-316](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-316.md)
+
+> **Note:**  MetaMask treats `requiredScopes` as `optionalScopes`. Only `optionalScopes` are recommended, though `requiredScopes` can be used to signal that your dapp will not be usable if certain [CAIP-217](https://chainagnostic.org/CAIPs/caip-217) `scopeStrings` are not authorized. 
+
+### CAIP-27 - Invoke RPC Requests for an Authorization Scope
+A dapp can invoke RPC requests for an authorization scope by making [CAIP-27](https://chainagnostic.org/CAIPs/caip-27) `wallet_invokeMethod` calls.
+
+> **Note:** a `sessionId` parameter is not supported and will be ignored, if included.
+
+### CAIP-312 - Retrieve Authorization Scopes
+A dapp can retrieve a [CAIP-312](https://chainagnostic.org/CAIPs/caip-312) multichain session object by calling `wallet_getSession` to request its authorization scopes at any time.
+
+> **Note:** a `sessionId` parameter is not supported and will be ignored, if included.
+
+### CAIP-311 - Notification of Changes to Authorization Scopes
+A dapp can listen for [CAIP-311](https://chainagnostic.org/CAIPs/caip-311) `wallet_sessionChanged` events in order to get notified when there are changes to its authorization scopes.
+
+> **Note:** A `sessionId` will not be included as part of the event.
+
+### CAIP-285 - Revoke Authorization Scopes
+A dapp can make a [CAIP-285](https://chainagnostic.org/CAIPs/caip-285) `wallet_revokeSession` request to revoke all of its authorization scopes.
+
+> **Note:** a `sessionId` parameter is not supported and will be ignored, if included.
+
+### API Specification Document
+An OpenRPC [specification for the Multichain API](https://github.com/MetaMask/api-specs/blob/main/multichain/openrpc.yaml) can be found in the [api-specs](https://github.com/MetaMask/api-specs) repository.
+
+## Caveats
+The CAIPs referenced in this proposal are still in Draft or Review status with CASA and may be subject to material changes.
+
+### API Access
+As part of the Multichain API, MetaMask is leading the adoption of alternative methods for browser extension wallet communication. The current practice for browser extension wallets is to inject a javascript object in order to communicate with each site. But this approach comes with privacy, security, reliability, and performance drawbacks.
+
+- The MetaMask Browser Extension will adopt an alternative method of communication through the `externally_connectable` feature, which is now supported by most modern web browsers with extension frameworks.
+- When a site is accessed through MetaMask's Mobile In-app Browser, the Multichain API will be available through an injected `window.wallet` object. 
```

Comment:
guessing this is placeholder, as in we haven't actually decided on the details of this yet

Reply:
yes, placeholder for now.

Reply:
Todo: Add a note about Firefox & Mobile.